Pilot/dirac-install.py

```diff
@@ -2192,7 +2192,9 @@ def createBashrc():
       if cliParams.userEnvVariables:
         lines.extend(['# User-requested variables'])
         for envName, envValue in cliParams.userEnvVariables.items():
-          lines.extend(['export %s=%s' % (envName, envValue)])
+          lines.extend(['( echo $%s | grep -q $%s ) || export %s=$%s:$%s' % (
+              envName, envValue,
+              envName, envName, envValue)])
```

Comment:
I'm a bit dubious about this line: I think it could do with a comment explaining what it does.

From what I can infer, I think it has an extra $ before the last %s... It may also fail if the variable already contains a superset of the value (I'm thinking of, for example, adding /bin if the variable already contains /usr/bin although that's probably a rarer case). Will it also add an extra ":" in if the variable isn't set in the first place?

Reply:
This line is not different from others in the same function, e.g.
```
( echo $PATH | grep -q $DIRACBIN ) || export PATH=$DIRACBIN:$PATH
```
This would end up adding
```
( echo $envName | grep -q $envValue ) || export envName=$envName:$envValue
```
Is it wrong?
Or maybe it should be 
```
( echo $envName | grep -q $envValue ) || export envName=$envValue:$envName
```
?

Reply:
Hmm, isn't envValue an actual value in this instance rather than another env variable? So it should be:
export envName=$envName:envValue (without the $ before the value).

Although this only works properly in cases where these things are lists of paths and already have a value set (so it's fine for PATH and probably LD_LIBRARY_PATH, but not general to other variables.

I think the original behaviour where it just set the value may be closer to what an end user expects when they ask for a variable to be set.

Reply:
I am removing the changes from this file until we sort out for https://github.com/DIRACGrid/DIRAC/issues/4860#issuecomment-754585681

Note that right now the changes highlighted here are the same already merged in DIRAC repo's dirac-install.py

Reply:
@sfayer please review now